lib/backend/buffer.go

```diff
@@ -280,12 +294,38 @@ func (w *BufferWatcher) Done() <-chan struct{} {
 }
 
 // Close closes the watcher, could
-// be called multiple times
+// be called multiple times, removes the watcher
+// from the buffer queue
 func (w *BufferWatcher) Close() error {
+	w.closeAndRemove(removeAsync)
+	return nil
+}
+
+// closeWatcher closes watcher
+func (w *BufferWatcher) closeWatcher() error {
 	w.cancel()
 	return nil
 }
 
+const (
+	removeSync  = true
+	removeAsync = true
```

Comment:
I'm guessing this should actually be `false` (judging by how it's used and the function below).

Reply:
:+1: good catch!